x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/AutoscalingIT.java

```diff
@@ -0,0 +1,194 @@
+import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingDeciderResults;
+import org.elasticsearch.xpack.core.ml.job.config.AnalysisConfig;
+import org.elasticsearch.xpack.core.ml.job.config.AnalysisLimits;
+import org.elasticsearch.xpack.core.ml.job.config.DataDescription;
+import org.elasticsearch.xpack.core.ml.job.config.Detector;
+import org.elasticsearch.xpack.core.ml.job.config.Job;
+import org.elasticsearch.xpack.datastreams.DataStreamsPlugin;
+import org.elasticsearch.xpack.ilm.IndexLifecycle;
+import org.elasticsearch.xpack.ml.LocalStateMachineLearning;
+import org.elasticsearch.xpack.ml.autoscaling.MlAutoscalingDeciderService;
+import org.elasticsearch.xpack.ml.autoscaling.MlScalingReason;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+
+public class AutoscalingIT extends MlNativeAutodetectIntegTestCase {
+
+    @Override
+    protected List<NamedWriteableRegistry.Entry> getNamedWritableEntries() {
+        List<NamedWriteableRegistry.Entry> entries = super.getNamedWritableEntries();
+        entries.add(new NamedWriteableRegistry.Entry(Metadata.Custom.class, AutoscalingMetadata.NAME, AutoscalingMetadata::new));
+        entries.add(new NamedWriteableRegistry.Entry(NamedDiff.class,
+            AutoscalingMetadata.NAME,
+            AutoscalingMetadata.AutoscalingMetadataDiff::new));
+        entries.add(new NamedWriteableRegistry.Entry(
+            AutoscalingDeciderResult.Reason.class,
+            MlScalingReason.NAME,
+            MlScalingReason::new
+        ));
+        return entries;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(
+            LocalStateMachineLearning.class,
+            Netty4Plugin.class,
+            ReindexPlugin.class,
+            Autoscaling.class,
+            // The monitoring plugin requires script and gsub processors to be loaded
+            IngestCommonPlugin.class,
+            // The monitoring plugin script processor references painless. Include this for script compilation.
+            // This is to reduce log spam
+            MockPainlessScriptEngine.TestPlugin.class,
+            // ILM is required for .ml-state template index settings
+            IndexLifecycle.class,
+            DataStreamsPlugin.class);
+    }
+
+    @Override
+    protected Settings externalClusterClientSettings() {
+        return Settings.builder().put(super.externalClusterClientSettings())
+            .put(Autoscaling.AUTOSCALING_ENABLED_SETTING.getKey(), true)
+            .build();
+    }
+
+    public void testMLAutoscalingCapacity() {
```

Comment:
This is running on the assumption that `xpack.ml.max_machine_memory_percent` is set to the default of `30` and `xpack.ml.use_auto_machine_memory_percent` is `false` right?.  I think it's worth adding a comment to say that these are the expectations for this test and it will need modifying if either of those defaults is ever changed.  It will make it clearer for future maintainers where the numbers have come from.